ui/apps/platform/src/utils/searchUtils.ts

```diff
@@ -310,3 +311,10 @@ export function addRegexPrefixToFilters(
 
     return modifiedFilter;
 }
+
+// Uses the generatePath function from react-router in addition to adding the query params
+export const generatePathWithQuery = (pathTemplate, params, queryParams) => {
+    const path = generatePath(pathTemplate, params);
```

Comment:
Minor request - can we use the recently created `safeGeneratePath` here instead? Although it is very unlikely that a code change would get through that changes the path template/params mapping in an inconsistent way, if somehow that happened `generatePath` would throw and crash the entire main page of the app.